### ISSUE TITLE
Uploading tracks doesn't work

### ISSUE THREAD

**Issue by JasonSanford** [fulcrumapp/fulcrum-js#20]
The audio and video members have an `uploadTrack` method that looks like it is broken. The current signature is `client.audio.uploadTrack(file, attributes)` where `attributes` is required to have an `access_id` key with a new or current photo/video id.

This seems kinda funky, and since it's been broken since v2 it might make sense to make the method signature `client.audio.uploadTrack(file, id?)` where id is optional and automatically created if not supplied.

**Comment by bmcbride** [comment]
Looks like the `fulcrum-ruby` library includes tracks as an option in the media `create` method: https://github.com/fulcrumapp/fulcrum-ruby#clientvideoscreatefile_or_path-content_type--videomp4-attributes--

**Comment by JasonSanford** [comment]
Yeah, that seems like a reasonable approach. If we go that route we could possibly just remove the `createTrack` method since it doesn't work and would be covered by this.

It might look something like:

```javascript
import fs from 'fs';
import uuid from 'uuid';

const video = fs.createReadStream('video.mp4');
const track = fs.createReadStream('track.json');
const key = uuid.v4();

const options = {
  accessKey: key,
  track: track
};

client.videos.create(video, options)
  .then(created => console.log(created))
  .catch(error => console.log(error));
```

**Comment by bmcbride** [comment]
I like that, but what happens if you want to attach a track to an existing video?

**Comment by JasonSanford** [comment]
I guess that was the idea behind the `uploadTrack` method? It assumed you would pass the id via `access_key` attribute: https://github.com/fulcrumapp/fulcrum-js/blob/df007e6fd73ae94fb3267c5e9bac63784ce7eb41/src/actions/track.js#L8

**Comment by bmcbride** [comment]
Maybe we need both

**Comment by JasonSanford** [comment]
Fixed in #23.
